toonz/sources/toonz/xshcolumnviewer.cpp

```diff
@@ -1315,13 +1334,38 @@ m_value->setFixedWidth(30);
 static QFont font("Helvetica", 7, QFont::Normal);
 m_value->setFont(font);*/
 
-  QHBoxLayout *hlayout = new QHBoxLayout;
-  hlayout->setContentsMargins(0, 3, 0, 3);
-  hlayout->setSpacing(1);
-  hlayout->addWidget(m_slider);
-  hlayout->addWidget(m_value);
-  hlayout->addWidget(new QLabel("%"));
-  setLayout(hlayout);
+  m_filterColorCombo = new QComboBox(this);
+  m_filterColorCombo->addItem(tr("None"), 0);
+  m_filterColorCombo->addItem(getColorChipIcon(1), tr("Red"), 1);
+  m_filterColorCombo->addItem(getColorChipIcon(2), tr("Green"), 2);
+  m_filterColorCombo->addItem(getColorChipIcon(3), tr("Blue"), 3);
+  m_filterColorCombo->addItem(getColorChipIcon(4), tr("DarkYellow"), 4);
+  m_filterColorCombo->addItem(getColorChipIcon(5), tr("DarkCyan"), 5);
+  m_filterColorCombo->addItem(getColorChipIcon(6), tr("DarkMagenta"), 6);
+
+  QVBoxLayout *mainLayout = new QVBoxLayout();
+  mainLayout->setMargin(3);
+  mainLayout->setSpacing(3);
+  {
+    QHBoxLayout *hlayout = new QHBoxLayout;
+    // hlayout->setContentsMargins(0, 3, 0, 3);
+    hlayout->setMargin(0);
+    hlayout->setSpacing(1);
+    hlayout->addWidget(m_slider);
+    hlayout->addWidget(m_value);
+    hlayout->addWidget(new QLabel("%"));
+    mainLayout->addLayout(hlayout, 0);
+
+    QHBoxLayout *filterColorLay = new QHBoxLayout();
+    filterColorLay->setMargin(0);
+    filterColorLay->setSpacing(2);
+    {
+      filterColorLay->addWidget(new QLabel(tr("Filter:"), this), 0);
```

Comment:
Thank you very much for catching this. I forgot to give thought to the case of vector images.
I'll modify as you suggested and will cover vector levels in the future, as another PR.